mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/IndexedRowMatrix.scala

```diff
@@ -113,6 +114,67 @@ class IndexedRowMatrix @Since("1.0.0") (
   }
 
   /**
+    * Converts to BlockMatrix. Creates blocks of `DenseMatrix` with size 1024 x 1024.
+    */
+  def toBlockMatrixDense(): BlockMatrix = {
```

Comment:
I am not sure if DenseMatrix-backed is better default behavior for `toBlockMatrix`. Actually the rows in `IndexedRowMatrix` can be sparse or dense. Choose which one, SparkMatrix-backed or DenseMartix-backed, is totally depending on the use case.

Looks like `toBlockMatrixDense` is already a non-small function. Merging it with current `toBlockMatrix` might not a good idea. I'd keep it as it's now.